src/main/scala/com/example/Service/ReservationService.scala

```diff
@@ -0,0 +1,148 @@
+package com.example.Service
+
+import java.sql.Timestamp
+
+import akka.actor.typed.scaladsl.Behaviors
+import akka.actor.typed.{ActorRef, Behavior}
+import akka.pattern.StatusReply
+
+import scala.util.{Failure, Success}
+import com.example.Registry.Registry.{CreateReservationDB, GetReservationsScreeningDB, GetScreeningScreeningRoomDB, GetScreeningTimeDB}
+import com.example.Registry.{Registry, ReservationCreatedDB, ReservationsScreeningDB, ScreeningScreeningRoomDB, ScreeningTime}
+
+import scala.concurrent.duration._
+import akka.util.Timeout
+import pureconfig.ConfigSource
+
+import scala.util.matching.Regex
+import pureconfig.generic.auto._
+
+final case class Reservation(screeningId: Int, name: String, seats: Seq[(Int, Int)], adult_tickets: Int, student_tickets: Int, child_tickets: Int)
+final case class ReservationResponse(total_amount: String, expiration_time: Timestamp)
+
+final case class Prices(adultTicketsPrice: Double, studentTicketsPrice: Double, childTicketsPrice: Double)
+final case class ExpirationTime(expirationTime: Long)
+
+object ReservationService {
+  sealed trait ReservationServiceCommand
+  final case class CreateReservation(reservation: Reservation, replyTo: ActorRef[Either[ReservationServiceError, ReservationResponse]]) extends ReservationServiceCommand
+
+  final case class GetReservationsScreeningDBResponse(reservedSeats: Seq[(Int, Int)], reservation: Reservation, replyTo: ActorRef[Either[ReservationService.ReservationServiceError, ReservationResponse]]) extends ReservationServiceCommand
+  final case class GetScreeningRoomDBResponse(reservation: Reservation, replyTo: ActorRef[Either[ReservationServiceError, ReservationResponse]]) extends ReservationServiceCommand
+  final case class GetScreeningTimeDBResponse(reservation: Reservation, time: Timestamp, replyTo: ActorRef[Either[ReservationServiceError, ReservationResponse]]) extends ReservationServiceCommand
+
+  final case class CreateReservationDBResponse(response: Either[ReservationServiceError, (Reservation, Timestamp)], replyTo: ActorRef[Either[ReservationServiceError, ReservationResponse]]) extends ReservationServiceCommand
+
+  trait ReservationServiceError
+  case class ExceptionError(error: String) extends ReservationServiceError
+  case class NotFoundError(error: String) extends ReservationServiceError
+  case class WrongParameter(error: String) extends ReservationServiceError
+
+  def apply(registry: ActorRef[Registry.RegistryCommand]): Behavior[ReservationServiceCommand] = {
+    reservationService(ConfigSource.default.loadOrThrow[Prices], ConfigSource.default.loadOrThrow[ExpirationTime], registry)
+  }
+
+  private def reservationService(prices: Prices, expirationTime: ExpirationTime, registry: ActorRef[Registry.RegistryCommand]): Behavior[ReservationServiceCommand] = {
+    Behaviors.setup[ReservationServiceCommand] { context =>
+      implicit val timeout: Timeout = 3.seconds
```

Comment:
is it ok to hardcode timeout here? why or why not?

Reply:
having hardcoded values is bad practice, it should be taken from config file